runtime/runtime/shutdown.go

```diff
@@ -0,0 +1,76 @@
+package runtime
+
+import (
+	"context"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	"encore.dev/runtime/config"
+)
+
+var shutdown = struct {
+	initiated chan struct{} // closed when graceful shutdown is initiated
+	completed chan struct{} // closed when graceful shutdown is completed
+	once      sync.Once     // to trigger shutdown logic only once
+
+	mu       sync.Mutex
+	handlers []func(force context.Context)
+}{
+	initiated: make(chan struct{}),
+	completed: make(chan struct{}),
+}
+
+func RegisterShutdown(fn func(force context.Context)) {
```

Comment:
Could you document this func please.

It’s not clear what the “force” context is for or why it’s called force

Reply:
Good point, will do